src/TriggerBinding/SqlTriggerTargetScaler.cs

```
@@ -14,24 +17,71 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
+                    // If we have already tried, we will return 0 as the target worker count.
+                    this._logger.LogWarning("Invalid object name detected. SQL trigger tables not found.");
+                    if (_lastTableCheck != DateTime.MinValue)
+                    {
+                        // Check if there are any changes in the user table.
+                        int changes = await GetNumberOfChangesAsync(this._connectionString, this._userTable, this._logger, CancellationToken.None);
+                        // If there are changes in the user table, we need to spin up a worker to create the global state and leases table.
+                        if (changes > 0)
+                        {
+                            this._logger.LogWarning("There are changes in the change-tracking table for the user table, but the global state and leases table are not created. Spinning up a worker to create those tables.");
+                            _lastTableCheck = DateTime.UtcNow;
+                            return new TargetScalerResult
+                            {
+                                TargetWorkerCount = 1
+                            };
+                        }
+                    }
 
-            return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+                }
+                // If the exception is not related to the invalid object name Or if there are no changes in the change tracking table for the user table yet.
+                // We can safely return 0 as the target worker count, indicating that no workers are needed at this time.
+                this._logger.LogWarning("An error occurred while getting the scale result for SQL trigger. Returning 0 as the target worker count. Exception: {ExceptionMessage}", ex.Message);
+                return new TargetScalerResult
+                {
+                    TargetWorkerCount = 0
```

@alrod If an exception occurs while getting the metrics count would it be better to let the exception flow back to the scale controller? Or is having it scale down to 0 until the issue is resolved expected behavior?

---

yeah, we should throw the exception back to SC here.